### PR TITLE
Fix mock migration

### PR DESCRIPTION
**Pull request by nazavode**
Misc fixes to have mock migration working. Now users, namespace and project will be created if not already present in GitLab database.

### CODE REVIEW

**Comment by glensc** [comment]
regarding users creation, do they get email sent out? and what is their initial state? active? blocked?

as if i created (disabled) users manually i encountered such issue:
https://gitlab.com/gitlab-org/gitlab-ce/issues/17487

as if i'm importing i will have lots of dead users whose accounts need to be created and their accounts disabled. meaning also that the emails the users have are already invalid.

File: tracboat/gitlab/__init__.py [hunk 1]
```diff
@@ -1 +1,93 @@
...
+
+
+@six.add_metaclass(abc.ABCMeta)
+class ConnectionBase(object):
+
+    def __init__(self, project_name):
+        project = project_name.strip()
+        if not project:
+            raise ValueError('invalid project name: {!r}'.format(project_name))
+        namespace, project = split_project_components(project)
```
Comment (glensc) [inline tracboat/gitlab/__init__.py:25]: gitlab 9 has nested groups support! this may break if enabled.

Comment (nazavode) [reply tracboat/gitlab/__init__.py:25]: Thanks, I was not aware of nested groups, I have to figure out how are they stored in the database schema, I'm going to try with GitLab 9.0.0...

**Comment by nazavode** [comment]
About users creation, I still have to figure out how to handle it gracefully, but for an initial release I was thinking about creating accounts in 'active' state and sending out simple notification emails just to inform the user an account has been created, of course at the first login he must go through a password reset procedure since I don't want to send generated passwords around.

**Comment by glensc** [comment]
i want users be created with state = blocked; type = external user. so perhaps make it configurable. altho i planned to create all missing users manually, until i saw you implemented user creation from the tool itself.

and if using database direct mode, then likely the users will be created silently (no mails being sent out). i.e the gitlab bug i referred does not apply.

**Comment by nazavode** [comment]
@glensc opened #15 for this, makes a lot of sense to me

**Pull request merged**